### DISCUSSION TITLE
Is it possible to postpone missed replication target ?

### DISCUSSION THREAD

**Discussion by mulder999** [seaweedfs/seaweedfs#5192] [General]
I've configured the default replication as 010, but I've noticed that it prevents any storage operation when the second rack is temporarily unavailable.

While I understand that this simplification might have facilitated the implementation, I would anticipate a distributed storage system to handle such scenarios by queuing operations for later execution to ensure the replication goal is eventually met.

Without this capability, the system lacks the resilience to tolerate transient failures and may face operational challenges in maintaining data consistency and availability during temporary disruptions with one of the rack.

**Comment by chrislusf** [comment]
add more racks.

**Reply by ruslan-mogilevskiy** [reply]
@mulder999 , the 010 replication will keep 2 copies of your data, one of which will be saved on a rack other to where the initial (first) copy is saved. So you need at least 2 (according to 010 replication) + 1 (extra) = **3** racks for cluster resiliency, and only one rack can go offline in this schema, otherwise (if more racks are down) the cluster will deny all write operations.

**Comment by mulder999** [comment]
Thanks for answers. So that would also implies 3 fully independant networks, is there a more cost effective way to achieve a hot spare scenario with automatic failover ?

**Reply by mulder999** [reply]
Replying to myself, I think I would rather need 2 clusters for such scenario: [Filer-Active-Active-cross-cluster-continuous-synchronization](https://github.com/seaweedfs/seaweedfs/wiki/Filer-Active-Active-cross-cluster-continuous-synchronization). However this starts to get really over complex and I should probably stick with glusterfs for my simple basic needs.

**Comment by mulder999** [comment]
Closing the question as I have decided to dismiss seaweedfs for my current project, as its actual implementation doesn't align with my practical requirements. For those who have come across this decision, I've opted for glusterfs in the replicated 2 arbiter 1 mode as my preferred solution. This configuration seamlessly handles scenarios where one node is down during read/write operations without any complications. Additionally, the storage space required for the arbiter is minimal, making it a very cost effective choice for my needs.
